specification.md

```diff
@@ -0,0 +1,131 @@
+
+**Data fetching** - it is network layer agnostic.
+Websockets or HTTP, REST, RPC or GraphQL - doesn't matter as long as the principles are not broken.
+
+## How
+
+Feature Driven Architecture defines foundational boundaries: Feature, Cluster, Screen, Shared and Library. Each concept follows the core principles.
+
+A concept of a "Feature" is the foundation of Feature Driven Architecture. Everything else exists to support it.
+
+Breaking any principle described in this architecture will lead to specific problems in your codebase, and the goal here is to define the benefits and trade-offs.
+
+## Core principles
+
+1. Decentralization
+1. Co-location
+1. Isolation
+1. Disposability
+1. Explicit sharing
+
+## Feature
+
+> A Feature is an isolated user-facing reusable complex building block(s) that solves a particular problem for a user.
+
+Feature scope generally depends on the product and organizational needs, but a specific Feature should have a clearly defined scope by its maintainers.
+
+[Read more about how to identify a feature.](./feature.md)
+
+A Feature follows these principles:
+
+1. A Feature can not know about the existence of any other Feature.
+
+   This principle is **the most important** one, and without it - your entire architecture is considered broken.
+
+1. A Feature can not know about the Screen it's been used on.
+
+   Without it, you won't be able to reuse a Feature across multiple screens.
+
+1. A Feature must be cohesive.
+
+   A Feature must co-locate its implementation in the location associated with the Feature name.
+
+1. A Feature can have dependencies.
+
+   A Feature must express dependencies in a form designed to be understood and provided by its consumers.
+
+1. A Feature has to be disposable.
+
+   You should be able to remove a Feature without leaving any remains behind.
+
+1. A Feature has to be composable.
+
+   You should be able to compose multiple Features on the same or different screens without modifying a Feature.
+
+## Screen
+
+> A Screen represents the content of the entire Screen.
+
+A Screen can use Clusters and Features to position them on the Screen. A Screen is responsible for the global layout. A Screen knows all Clusters and Features it is using.
+
+A Screen follows these principles:
+
+1. A Screen can not know about the existence of any other Screen.
+1. A Screen must be cohesive.
+1. A Screen has to be disposable.
+
+## Cluster
+
+> Cluster is a reusable bundle of Features.
+
+A Cluster is useful when you need to bundle several Features and reuse them together on different Screens.
+
+A Cluster can be used to forward data from one Feature to another, and to configure Features.
+
+A Cluster is needed to avoid duplication of setup code for Features. If the setup is trivial - you don't need it. _Cluster is optional._
+
+A Cluster follows these principles:
+
+1. A Cluster can not know about the existence of any other Cluster.
+1. A Cluster can not know about the Screen it's been used on.
```

Comment:
More of a curiosity but where does Screen-specific logic live in this architecture/who manages Cluster/Feature dynamism? 
If a Features in a Cluster being used by `Screen A` should look/behave differently on `Screen B`, is that the responsibility of the Screen?

Reply:
Yeah, it would  mean a feature or cluster has to be configurable and used with different config from different screens